test/services/mail_service_test.dart

```diff
@@ -0,0 +1,130 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:summer2022/exceptions/fetch_mail_exception.dart';
+import 'package:summer2022/services/mail_service.dart';
+
+void main() async {
+  final search = MailService();
+
+  DateTime now = DateTime.now();
+
+  DateTime today = new DateTime(now.year, now.month, now.day);
+
+  List<MailPiece> mail = <MailPiece>[
+    new MailPiece("", "", "", "", "test", today),
+    new MailPiece("", "", "", "", "", DateTime.now().add(Duration(days: 1)))
+  ];
+
+  test("Search mail by keyword", () {
+    String error = '';
+    try {
+      List<MailPiece> test = search.getMailByKeyword(mail, "test");
+      expect(1, test.length);
+    } catch (e) {
+      error = e.toString();
+    }
+    expect(error, '');
```

Comment:
Same comment for all tests:

I think you can simplify this by removing the `error` variable and the `try/catch`. If the test throws an exception, it should automatically fail the test.